### PR TITLE
refactor(auth): remove RelativeLayout usage

### PR DESCRIPTION
**Pull request by rosariopfernandes**
See discussion in #1217 

**~This is a Work-In-Progress:~**
### Anonymous Auth
<table><thead><tr><th>Before</th><th>After</th></tr></thead><tbody><tr><td>

</td><td>

</td></tr></tbody></table>

### Email Password
<table><thead><tr><th>Before</th><th>After</th></tr></thead><tbody><tr><td>

</td><td>

</td></tr></tbody></table>

### Facebook
<table><thead><tr><th>Before</th><th>After</th></tr></thead><tbody><tr><td>

</td><td>

</td></tr></tbody></table>

### Firebase UI
<table><thead><tr><th>Before</th><th>After</th></tr></thead><tbody><tr><td>

</td><td>

</td></tr></tbody></table>

### Generic IDP
<table><thead><tr><th>Before</th><th>After</th></tr></thead><tbody><tr><td>

</td><td>

</td></tr></tbody></table>

### Google
<table><thead><tr><th>Before</th><th>After</th></tr></thead><tbody><tr><td>

</td><td>

</td></tr></tbody></table>

### Multi Factor Auth

I changed this one to give it a better look because the buttons were all squeezed in there.

<table><thead><tr><th>Before</th><th>After</th></tr></thead><tbody><tr><td>

</td><td>

</td></tr></tbody></table>

### Passwordless
<table><thead><tr><th>Before</th><th>After</th></tr></thead><tbody><tr><td>

</td><td>

</td></tr></tbody></table>

### Phone Auth
<table><thead><tr><th>Before</th><th>After</th></tr></thead><tbody><tr><td>

</td><td>

</td></tr></tbody></table>

### CODE REVIEW

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PasswordlessActivity.kt [hunk 1]
```diff
@@ -195,14 +195,11 @@ class PasswordlessActivity : BaseActivity(), View.OnClickListener {
         if (user != null) {
             binding.status.text = getString(R.string.passwordless_status_fmt,
                     user.email, user.isEmailVerified)
-
-            binding.passwordlessFields.visibility = View.GONE
             binding.passwordlessButtons.visibility = View.GONE
-            binding.signedInButtons.visibility = View.VISIBLE
+            binding.signOutButton.visibility = View.VISIBLE
```
Comment (samtstern) [inline auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PasswordlessActivity.kt:199]: Is this the right change?  We used to hide a group of buttons now you're just hiding one.

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PasswordlessActivity.kt [hunk 2]
```diff
@@ -195,14 +195,11 @@ class PasswordlessActivity : BaseActivity(), View.OnClickListener {
...
         } else {
-            binding.passwordlessFields.visibility = View.VISIBLE
             binding.passwordlessButtons.visibility = View.VISIBLE
-            binding.signedInButtons.visibility = View.GONE
+            binding.signOutButton.visibility = View.GONE
```
Comment (samtstern) [inline auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PasswordlessActivity.kt:202]: Same question here.

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PhoneAuthActivity.kt [hunk 3]
```diff
@@ -287,13 +287,13 @@ class PhoneAuthActivity : AppCompatActivity(), View.OnClickListener {
         if (user == null) {
             // Signed out
             binding.phoneAuthFields.visibility = View.VISIBLE
-            binding.signedInButtons.visibility = View.GONE
+            binding.signOutButton.visibility = View.GONE
```
Comment (samtstern) [inline auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PhoneAuthActivity.kt:290]: Same here

**Review by samtstern — APPROVED** [review]
Approved with some small questions about changes to button groups.

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PasswordlessActivity.kt (hunk 1 above)
Comment (rosariopfernandes) [reply auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PasswordlessActivity.kt:199]: Yes, the change is correct. That group was a LinearLayout containing only a single button, so I removed it and kept the button only.

https://github.com/firebase/quickstart-android/blob/368ff73c206bb767672118f22aed0254a7dcb473/auth/app/src/main/res/layout/activity_passwordless.xml#L110-L130

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PhoneAuthActivity.kt (hunk 3 above)
Comment (rosariopfernandes) [reply auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PhoneAuthActivity.kt:290]: Addressed above, but I double checked:

https://github.com/firebase/quickstart-android/blob/368ff73c206bb767672118f22aed0254a7dcb473/auth/app/src/main/res/layout/activity_phone_auth.xml#L121-L140

**Comment by samtstern** [comment]
@rosariopfernandes makes sense, thank you!  Also this PR taught me a lot about Guidelines and Groups in ConstraintLayout :-)

**Comment by rosariopfernandes** [comment]
@samtstern I'm glad :)

I noticed in your RTDB PR that we could've used a guideline (in activity_sign_in.xml) to center the textfields and replace their fixed width with `match_constraint` (0dp). I didn't comment on it because the previous Layout was also using fixed width, so I thought we could just leave as is. But if you want to try out Guidelines, I think that'd be a good place to do so ;)

**Pull request merged**